### DISCUSSION TITLE
@next/bundle-analyzer can't accept an async configuration function

### DISCUSSION THREAD

**Discussion by Janpot** [vercel/next.js#43759] [Ideas]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```
    Operating System:
      Platform: linux
      Arch: x64
      Version: Ubuntu 20.04.0 LTS Mon Dec 05 2022 17:39:31 GMT+0100 (Central European Standard Time)
    Binaries:
      Node: 16.14.2
      npm: 7.17.0
      Yarn: 1.22.19
      pnpm: 7.13.6
    Relevant packages:
      next: 13.0.7-canary.1
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0
```

### Link to reproduction - Issues with a link to complete (but minimal) reproduction code will be addressed faster

 https://stackblitz.com/edit/nextjs-6yqgx4?file=next.config.mjs

### To Reproduce

1. Run next in https://stackblitz.com/edit/nextjs-6yqgx4?file=next.config.mjs
2. Observe the console:

    ```
    { env: { FOO: 'bar' }, webpack: [Function: webpack] } { webpack: [Function: webpack] }
    ```

### Describe the Bug

`@next/bundle-analyzer` can't handle input when it's an async function. Sentry [returns a function now](https://github.com/getsentry/sentry-javascript/commit/f1e59f9973a092724615b7fbc421a893eb6438c3) and the output became incompatible with  `@next/bundle-analyzer`. e.g.

```ts
export default withBundleAnalyzer(withSentryConfig(myConfig, sentryConfig));
```
doesn't work anymore and

```ts
export default withSentryConfig(withBundleAnalyzer(myConfig), sentryConfig);
```
does

Related: https://github.com/getsentry/sentry-javascript/issues/6401

### Expected Behavior

`@next/bundle-analyzer` should accept a function as input

**Comment by balazsorban44** [comment]
Sounds like a feature request, should be handled here probably: https://github.com/vercel/next.js/blob/canary/packages/next-bundle-analyzer/index.js
